app/Http/Controllers/RepositoryController.php

```diff
@@ -32,10 +32,24 @@ public function store(StoreRepositoryRequest $request): RedirectResponse
     {
         $data = RepositoryData::from($request->validated());
 
-        $request
-            ->user()
-            ->repositories()
-            ->create($data->toArray());
+        try {
```

Comment:
It is better to avoid too much code logic in the controller.
could we move it to service?
and also could we use the update or create instead of try catch?

Reply:
I didn't use the `updateOrCreate` method because it doesn't work in our case. Indeed, this code could have worked:

```php
$request
    ->user()
    ->repositories()
    ->updateOrCreate([
        'node_id' => $data->nodeId,
    ], [
        ...$data->toArray(),
        'deleted_at' => null, // Force the repository to be restored if it was soft deleted.
    ]);
```

But the `repositories_node_id_full_name_unique` key also contains the full_name. This can change between the first time the repo is added and the second time (unlike the node_id which is really unique).

Use a `try/catch` here is the best solution imo since we can explicitly search if the node_id already exists and update the entry. The code is also clearer because we use the `restore()` method

Reply:
Also, I agree on the fact of not put too much logic into a controller, but on the other hand I am also in favor of not creating several dozen files to carry out a single action.

For example, here the `FormRequest` used for the validation logic adds unnecessary complexity when a simple `$request->validate([...])` here would have been sufficient.

Reply:
the updateOrCreate work if we add ->restore()

Reply:
![image](https://user-images.githubusercontent.com/51865639/233572717-e1d41375-411c-4d13-bf3a-b36255a3427b.png)

Query log :